### ISSUE TITLE
Focus is not moving to "line1" link separately, entire line is treated as one interactive element: A11y_MSSQL for VSCode_Execute Query_Keyboard

### ISSUE THREAD

**Issue by Swati1700** [microsoft/vscode-mssql#20359]
### Description

#A11yTCS; #A11ySev2; #Win32; #WCAG2.1.1; #Keyboard; #MSSQL for VSCode; #Win11; #A11yMAS; #DesktopApp; #Benchmark; #AILimited; #BM_MSSQL for VSCode_Win32_Oct2025; #Closed; #Regressed:11-12-25;

### Environment Information

Application Name: MSSQL for VSCode. 
Version: 1.104.3 (user setup)
OS Version: Windows 11 Enterprise 24H2 (OS Build 26100.4652) 

### Steps to Reproduce

"Please do not close this bug. This bug should only be closed by TCS, C+AI Accessibility tester after bug verification." 
 
"[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier." 

**Repro steps:** 

1.Launch Visual Studio Code Application. 
2.Install MSSQL Extension and connect to a SQL Server. 
3.Tab to Server present under "Connections Side Bar" and Press context menu key, it opens a list. 
4.Press enter on "New Query". 
5.Execute a valid query. 
6.Press Ctrl+M, Tab to exit from editor. Tab to "Message" section and press enter to view the contents. 
7.Observe whether focus is moving to "line1" link separately or not. 

**Actual Result:** 
The entire line is receiving focus, instead of allowing focus only on the “Line 1” link. 

**Note: 1.Similar issue observed for the below scenario. 

Repro steps:** 

1. Launch Visual Studio Code Application. 
2. Install MSSQL Extension and connect to a SQL Server. 
3. Tab to Server present under "Connections Side Bar" and press enter to expand. 
4. Continue expanding: System Databases -> Master -> Tables -> System Tables. 
5. Press context menu key to open the list. 
6. Press enter on “Select top 1000”. This will open an editor window with the query and display the results, if any. 
7. Tab till messages. 
8. Observe whether the “Line 1” link is focusable and accessible separately as an individual interactive element or not.

**2.Similar issue observed for the below scenario.** 
**Repro steps:** 

1.Launch Visual Studio Code Application. 
2.Install MSSQL Extension and connect to a SQL Server. 
3.Tab to Server present under "Connections Side Bar" and press enter to expand. 
4.Continue expanding: System Databases -> Master -> Tables -> System Tables. 
5.Press context menu key to open the list. 
6.Press enter on “Script as Create”. This will open an editor window with the query and display the results, if any. 
7.Tab till messages. 
8.observe whether the “Line 1” link is focusable and accessible separately as an individual interactive element or not 

**3.Similar issue observed for the below scenario. 
Repro steps:** 

1.Launch Visual Studio Code Application. 
2.Install MSSQL Extension and connect to a SQL Server. 
3.Tab to Server present under "Connections Side Bar" and press enter to expand. 
4.Continue expanding: System Databases -> Master -> Tables -> System Tables. 
5.Press context menu key to open the list. 
6.Press enter on “Script as Drop”. This will open an editor window with the query and display the results, if any. 
7.Tab till messages. 
8.observe whether the “Line 1” link is focusable and accessible separately as an individual interactive element or not 

**4.Similar issue observed for the below scenario. 
Repro steps:** 

1. Launch Visual Studio Code Application. 
2. Install MSSQL Extension and connect to a SQL Server. 
3. Tab to Server present under "Connections Side Bar" and press enter to expand. 
4. Continue expanding: System Databases -> Master -> views-> System views. 
5. Press context menu key to open the list. 
6. Press enter on “Script as Alter”. This will open an editor window with the query and display the results, if any. 
7. Tab till messages. 
8. Observe whether the “Line 1” link is focusable and accessible separately as an individual interactive element or not 

**Expected Result:** 
The “Line 1” link should be focusable and accessible separately as an individual interactive element. 

**User Impact:** 
Keyboard users cannot directly access the “line1” link, making navigation inefficient. 

**Recommendations:** 

Refer [Developer Resources (sharepoint.com)](https://microsoft.sharepoint.com/sites/accessibility/SitePages/Developer-Resources.aspx). 

**MAS Reference:** 

[MAS 2.1.1 – Keyboard](https://aka.ms/MAS2.1.1)  

Check out [MAS 2.1.1: Keyboard Guide](https://microsoft.sharepoint.com/sites/accessibility/SitePages/MAS-2.1.1-Keyboard-Guide.aspx)  for resources and suggestions.

Attachment:

https://github.com/user-attachments/assets/890f7b56-b23e-4dfd-b9e0-c3d22cd96496

Similar issue:  

[Similar issue.zip](https://github.com/user-attachments/files/23299622/Similar.issue.zip)

**Comment by laurenastrid1** [comment]
@Swati1700, this is the expected focus behavior based on Microsoft’s Fluent UI design.
In the Messages View, after pressing Tab to enter the grid, use the arrow keys to navigate between cells. In order to navigate in the cell, press enter, and then the link will be focused.

<img width="907" height="168" alt="Image" src="https://github.com/user-attachments/assets/a8ba5482-f222-4ffe-a63b-5280623c510d" />

Documentation: https://storybooks.fluentui.dev/react/?path=%2Fdocs%2Fcomponents-datagrid--docs

**Comment by Swati1700** [comment]
As per the above comment and after discussion with internal sme (Pawan) closing the bug.
